tuner/tuner/libtuner.py

```diff
@@ -696,6 +725,51 @@ def append_to_file(lines: list[str], filepath: Path, title: str = "") -> None:
         file.write("\n")
 
 
+def get_benchmark_context(benchmarks: Optional[dataclass]) -> dict:
+    """Gathers system information for the context section of the JSON output."""
+    cpu_info = psutil.cpu_freq()
+    num_cpus = psutil.cpu_count(logical=True)
+
+    benchmark_times = [benchmark.benchmark_time_in_seconds for benchmark in benchmarks]
+
+    if benchmark_times:
+        mean_time = sum(benchmark_times) / len(benchmark_times)
+    else:
+        mean_time = None
+
+    context = {
+        "date": datetime.now().isoformat(),
+        "num_cpus": num_cpus,
+        "mhz_per_cpu": cpu_info.max if cpu_info else "N/A",
+        "cpu_scaling_enabled": cpu_info is not None,
+        "mean_running_time": mean_time if mean_time is not None else "N/A",
+    }
+    return context
+
+
+def preprocess_benchmarks_for_serialization(benchmarks: list[dataclass]) -> list[dict]:
+    """Preprocess benchmarks to make them JSON serializable."""
+    preprocessed = []
+    for benchmark in benchmarks:
+        benchmark_dict = asdict(benchmark)
+        for key, value in benchmark_dict.items():
+            if isinstance(value, Path):
+                benchmark_dict[key] = str(value)
+        preprocessed.append(benchmark_dict)
+    return preprocessed
```

Comment:
Should I document this somewhere?

It seems quite ambiguous currently because one would need to know the context of the task or read through the code to know that the method is only for serializing benchmarks for json